### PR TITLE
http_dav.c: Can't check mailbox type when proxying PUT

### PR DESCRIPTION
**Pull request by ksmurchison**
Fixes a bug proxying PUT.  

Also adds a Cass test for proxying GET, PUT, MOVE, DELETE, PROPFIND, PROPPATCH, MKCALENDAR, REPORT (sync-collection, calendar-multiget).

Still need to add tests for proxying MOVEs across backends, but I don't think we allow a user's calendars to be split across backends.  We could do this with shared calendars across users on different backends, but DAV sharing across backends (the notifications part) isn't supported either.

@elliefm Do you want to backport the fix and tests?

### CODE REVIEW

**Review by elliefm — APPROVED** [review]
Looks reasonable.  One nit in the test, but maybe it's better as is

**Comment by elliefm** [comment]
@ksmurchison did you test this with a multi-version cassandane setup?

Marking this to backport as far back as 3.4 for now, because it's small and looks like it'll probably go smoothly.  If it doesn't go smoothly, I'll stop backporting where it starts hurting.

File: cassandane/Cassandane/Cyrus/MurderDAV.pm [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+</D:propfind>
+EOF
+
+    xlog $self, "Get current-user-principal";
+    my $url = $frontend_caldav->GetCurrentUserPrincipal();
+    $self->assert_not_null($url);
+
+    xlog $self, "PROPFIND for home-sets";
+    my (undef, undef, undef, $cur_princ) =
+        $url =~ m{^http(s)?://([^/:]+)(?::(\d+))?(.*)?};
```
[the remark this answers is not in the document]
Comment (ksmurchison) [reply cassandane/Cassandane/Cyrus/MurderDAV.pm:146]: I added a comment that it was copied from Net::DAVTalk::SetURL

**Comment by ksmurchison** [comment]
@elliefm I didn't test with a multi-version setup because I don't think it matters.  Standalone/backend PUT has been working since day one.  The only issue is that a frontend was doing a bogus check

**Pull request merged**